vnext/Shared/Modules/BlobModule.cpp

```diff
@@ -0,0 +1,538 @@
+               auto message = "Invalid type for blob: " + type;
+               Modules::SendEvent(state->Module->getInstance(), "blobFailed", std::move(message));
+             }
+             return;
+           }
+         }
+
+         persistor->StoreMessage(std::move(buffer), std::move(blobId));
+       }},
+
+      {"release",
+       [persistor = m_blobPersistor](dynamic args) // blobId: string
+       {
+         auto blobId = jsArgAsString(args, 0);
+
+         persistor->RemoveMessage(std::move(blobId));
+       }}};
+}
+
+#pragma endregion CxxModule
+
+#pragma endregion BlobModule
+
+#pragma region MemoryBlobPersistor
+
+#pragma region IBlobPersistor
+
+winrt::array_view<uint8_t> MemoryBlobPersistor::ResolveMessage(string &&blobId, int64_t offset, int64_t size) noexcept {
+  scoped_lock lock{m_mutex};
+
+  auto &data = m_blobs.at(std::move(blobId));
+
+  return winrt::array_view<uint8_t>{data};
+}
+
+void MemoryBlobPersistor::RemoveMessage(string &&blobId) noexcept {
+  scoped_lock lock{m_mutex};
+
+  m_blobs.erase(std::move(blobId));
+}
+
+void MemoryBlobPersistor::StoreMessage(vector<uint8_t> &&message, string &&blobId) noexcept {
+  scoped_lock lock{m_mutex};
+
+  m_blobs.insert_or_assign(std::move(blobId), std::move(message));
+}
+
+string MemoryBlobPersistor::StoreMessage(vector<uint8_t> &&message) noexcept {
+  // substr(1, 36) strips curly braces from a GUID.
+  auto blobId = winrt::to_string(winrt::to_hstring(GuidHelper::CreateNewGuid())).substr(1, 36);
+
+  scoped_lock lock{m_mutex};
+  m_blobs.insert_or_assign(blobId, std::move(message));
+
+  return blobId;
+}
+
+#pragma endregion IBlobPersistor
+
+#pragma endregion MemoryBlobPersistor
+
+#pragma region BlobWebSocketModuleContentHandler
+
+BlobWebSocketModuleContentHandler::BlobWebSocketModuleContentHandler(shared_ptr<IBlobPersistor> blobPersistor) noexcept
+    : m_blobPersistor{blobPersistor} {}
+
+#pragma region IWebSocketModuleContentHandler
+
+void BlobWebSocketModuleContentHandler::ProcessMessage(string &&message, dynamic &params) /*override*/ {
+  params[dataKey] = std::move(message);
+}
+
+void BlobWebSocketModuleContentHandler::ProcessMessage(vector<uint8_t> &&message, dynamic &params) /*override*/ {
+  auto blob = dynamic::object();
+  blob(offsetKey, 0);
+  blob(sizeKey, message.size());
+  blob(blobIdKey, m_blobPersistor->StoreMessage(std::move(message)));
+
+  params[dataKey] = std::move(blob);
+  params[typeKey] = blobKey;
+}
+#pragma endregion IWebSocketModuleContentHandler
+
+void BlobWebSocketModuleContentHandler::Register(int64_t socketID) noexcept {
+  scoped_lock lock{m_mutex};
+  m_socketIds.insert(socketID);
+}
+
+void BlobWebSocketModuleContentHandler::Unregister(int64_t socketID) noexcept {
+  scoped_lock lock{m_mutex};
+
+  auto itr = m_socketIds.find(socketID);
+  if (itr != m_socketIds.end())
+    m_socketIds.erase(itr);
+}
+
+#pragma endregion BlobWebSocketModuleContentHandler
+
+#pragma region BlobModuleUriHandler
```

Comment:
>BlobModuleUriHandler [](http://example.com/codeflow?start=15&length=20)

Looks like you have implemented assuming that the uri is a local filesystem path .. Is there a way to create custom url schemes on windows, akin to
iOS : [https://developer.apple.com/documentation/xcode/defining-a-custom-url-scheme-for-your-app](https://developer.apple.com/documentation/xcode/defining-a-custom-url-scheme-for-your-app)
Android : [https://developer.android.com/guide/topics/providers/content-providers](https://developer.android.com/guide/topics/providers/content-providers)

If not, the URI based blob resolution may not make sense on Windows .. One solution will be to use a temporary file based blob persister and use the file path as the URI though ..

Reply:
This would be a requirement for integrating blobs to ImageViewManager for instance.

Reply:
I did have that assumption.
I couldn't infer otherwise from the Android code base.

Following up.

Reply:
URI handlers have been postponed, left out of this PR.